tcms/bugs/tests/test_api.py

```diff
@@ -32,7 +31,9 @@ def verify_api_with_permission(self):
         self.assertIn(self.tag, self.bug.tags.all())
 
     def verify_api_without_permission(self):
-        with self.assertRaisesRegex(ProtocolError, "403 Forbidden"):
+        with self.assertRaisesRegex(
+            XmlRPCFault, 'Authentication failed when calling "Bug.add_tag"'
+        ):
```

Comment:
I would reconsider returning 200 on failure. It is non intuitive while a 403 or 500 is much more clear. This change has the potential to break existing clients which rely on the status code in order to perform some actions, e.g. a graceful error message. 

Our own RPC client is at https://github.com/kiwitcms/Kiwi/blob/master/tcms/static/js/jsonrpc.js#L23. I need to review in more details if the beckend change will not break the FE client but essentially we want to be able to show an alert to the user when something fails.

Reply:
Can you point me to the actual commit/commits or place in the code which implement this change? 

We do have quite a lot of API clients and I'd like to override this change and revert to the old behavior, e.g. by supplying a custom XMLRpcHandler. We've already got `KiwiTCMSXmlRpcHandler` so overriding a few more methods should be relatively easy.  Then we could think about phasing out the old behavior properly on our side.

Reply:
I am back to reviewing this PR. 

From what I can tell our RPC clients don't really depend on the 403 return code and after doing some testing it looks like they are working fine. I am going ahead with this change as-is.